agent/bench-scripts/driver/linpack

```diff
@@ -0,0 +1,277 @@
+	exit 5
+fi
+num_problem_sizes=$(echo "${problem_sizes}" | wc -w)
+num_leading_dimensions=$(echo "${leading_dimensions}" | wc -w)
+num_run_trials=$(echo "${run_trials}" | wc -w)
+num_alignment_values=$(echo "${alignment_values}" | wc -w)
+if [[ "${num_problem_sizes}" != "${num_leading_dimensions}" || "${num_leading_dimensions}" != "${num_run_trials}" || "${num_run_trials}" != "${num_alignment_values}" ]]; then
+	printf -- "[%s] ERROR: You must have the same number of elements for --problem-sizes, --leading-dimensions, --run-trials, and --alignment-values\n" >&2
+	exit 6
+fi
+number_of_tests=${num_problem_sizes}
+
+linpack_input_file="${output_dir}/${linpack_input_file_name}"
+linpack_metadata_file="${output_dir}/${linpack_metadata_file_name}"
+linpack_output_file="${output_dir}/${linpack_output_file_name}"
+
+numactl_cmd=""
+if [[ -n "${numactl_args}" ]]; then
+	# N.B. - Trailing whitespace is required.
+	numactl_cmd="numactl ${numactl_args} "
+fi
+
+if [[ "${use_omp}" == "y" ]]; then
+	if [[ -z "${threads_arg}" ]]; then
+		threads_arg=$(cat /proc/cpuinfo | grep processor | wc -l)
+	fi
+	# N.B. - Trailing whitespace is required.
+	omp_num_threads="OMP_NUM_THREADS=${threads_arg} "
+	# N.B. - Trailing whitespace is required.
+	kmp_affinity="KMP_AFFINITY=${kmp_affinity_args} "
+else
+	# When "use OpenMP" is false, we are not going to use the "threads_arg" or "kmp_affinity_args"
+	threads_arg=""
+	kmp_affinity_args=""
+
+	# No environment variables to add to the command line.
+	omp_num_threads=""
+	kmp_affinity=""
+fi
+final_cmd="${omp_num_threads}${kmp_affinity}${numactl_cmd}${binary} ${linpack_input_file} | stdbuf --input=0 tee -i ${linpack_output_file}"
+
+if [[ -z "${header}" ]]; then
+	# Line 1 default
+	header="Pbench - LINPACK Benchmark data file (${linpack_input_file})"
+fi
+if [[ -z "${subheader}" ]]; then
+	# Line 2 default header
+	subheader="Pbench LINPACK Benchmark data -- ${final_cmd}"
+fi
+
+# Create the LINPACK input file.
+printf -- "%s\n%s\n%d # number of tests\n%s # problem sizes\n%s # leading dimensions\n%s # times to run a test\n%s # alingment values (in KBytes)\n" \
+	"${header}" \
+	"${subheader}" \
+	"${number_of_tests}" \
+	"${problem_sizes}" \
+	"${leading_dimensions}" \
+	"${run_trials}" \
+	"${alignment_values}" \
+	> ${linpack_input_file}
+
+# Now we can execute the final LINPACK command.
+eval ${final_cmd}
+exit_code=${?}
+
+# Create the linpack.meta file containing all the metadata used for invoking
+# LINPACK (does not include "header" or "subheader").  This serves as the marker
+# file for its completion.
+printf -- "binary=%s\nuse_omp=%s\nthreads=%s\nkmp_affinity=%s\nnumactl_cmd=%s\nproblem_sizes=%s\nleading_dimensions=%s\nalignment_values=%s\n" \
+	"${binary}" \
+	"${use_omp}" \
+	"${threads_arg}" \
+	"${kmp_affinity_args}" \
+	"${numactl_cmd}" \
+	"${problem_sizes}" \
+	"${leading_dimensions}" \
+	"${alignment_values}" \
+	> ${linpack_metadata_file}
```

Comment:
I don't really like the long format string (it wraps...) nor the separation of the arguments from the format -- it makes the code hard to review.  I think it would be better if we let the shell expand the variables in-line, e.g.:
```
cat > ${linpack_metadata_file} <<-__EOF__
	binary=${binary}
	use_omp=${use_omp}
	threads=${threads_arg}
	kmp_affinity=${kmp_affinity}
	numactl_cmd=${numactl_cmd}
	problem_sizes=${problem_sizes}
	leading_dimensions=${leading_dimensions}
	alignment_values=${alignment_values}
	__EOF__
```
Something similar would be good at lines 249-257, although the need there isn't quite so great, because the lines are mostly just ordered, they aren't trying to match keys to values.